metomi/rose/loc_handlers/rsync.py

```diff
@@ -63,34 +63,11 @@ def parse(self, loc, _):
         cmd = self.manager.popen.get_cmd(
             "ssh", host, "python3", "-", path, loc.TYPE_BLOB, loc.TYPE_TREE)
         temp_file = TemporaryFile()
-        temp_file.write(br"""
-import os
-import sys
-path, str_blob, str_tree = sys.argv[1:]
-if os.path.isdir(path):
-    print(str_tree)
-    os.chdir(path)
-    for dirpath, dirnames, filenames in os.walk(path):
-        good_dirnames = []
-        for dirname in dirnames:
-            if not dirname.startswith("."):
-                good_dirnames.append(dirname)
-                name = os.path.join(dirpath, dirname)
-                print("-", "-", "-", name)
-        dirnames[:] = good_dirnames
-        for filename in filenames:
-            if filename.startswith("."):
-                continue
-            name = os.path.join(dirpath, filename)
-            stat = os.stat(name)
-            print(oct(stat.st_mode), stat.st_mtime, stat.st_size, name)
-elif os.path.isfile(path):
-    print(str_blob)
-    stat = os.stat(path)
-    print(oct(stat.st_mode), stat.st_mtime, stat.st_size, path)
-""")
+        temp_file.write(
+            (Path(__file__).parent / 'rsync_remote_check.py').read_bytes()
```

Comment:
Can import the path from the installed file directly

```python
from metomi.rose.loc_handlers.rsync_remote_check import __file__ as rsync_remote_check_file
# ...
Path(rsync_remote_check_file).read_bytes()
```

Reply:
Much tidier. Thanks.